Disable tracer warning for slicing indices.

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50414 Disable tracer warning for slicing indices.**

Summary: If the index that is supplied from python is an integral type, it converts everything to int64_t which is traced correctly. Therefore there is no need to warn the user that tracing might be wrong

Test Plan: new test case

Reviewers:

Subscribers:

Tasks: https://github.com/pytorch/pytorch/issues/49826

Tags:

Differential Revision: [D25930773](https://our.internmc.facebook.com/intern/diff/D25930773)

tugsbayasgalan committed:
Disable tracer warning for slicing indices.

Summary: If the index that is supplied from python is an integral type, it converts everything to int64_t which is traced correctly.

Test Plan: new test case

Reviewers:

Subscribers:

Tasks: https://github.com/pytorch/pytorch/issues/49826

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Disable tracer warning for slicing indices."


Summary: If the index that is supplied from python is an integral type, it converts everything to int64_t which is traced correctly.

Test Plan: new test case

Reviewers:

Subscribers:

Tasks: https://github.com/pytorch/pytorch/issues/49826

Tags:

[ghstack-poisoned]

wanchaol commented on test/jit/test_tracer.py:
```diff
@@ -150,6 +150,26 @@ def f(p):
+        test_inputs = (torch.randint(0, 9, (9, 9)), torch.tensor(8))
+        eager_out = mod(*test_inputs)
+        traced_out = traced_func(*test_inputs)
+        self.assertTrue(torch.allclose(eager_out, traced_out))
```
can you do a `assertNoWarn` check here? I recalled there should be some test utils you can check it, you can look at test_jit.py for reference.

tugsbayasgalan replied:
will do

tugsbayasgalan commented on tools/autograd/templates/python_variable_methods.cpp:
```diff
@@ -376,7 +376,7 @@ static PyObject * THPVariable_index_scalar(PyObject* self, PyObject* args) {
     return handle_torch_function(self, "__index__", args);
   }
-  jit::tracer::warn("Converting a tensor to a Python index", jit::tracer::WARN_PYTHON_DATAFLOW);
+  //jit::tracer::warn("Converting a tensor to a Python index", jit::tracer::WARN_PYTHON_DATAFLOW);
```
yep

tugsbayasgalan committed:
Update on "Disable tracer warning for slicing indices."


Summary: If the index that is supplied from python is an integral type, it converts everything to int64_t which is traced correctly.

Test Plan: new test case

Reviewers:

Subscribers:

Tasks: https://github.com/pytorch/pytorch/issues/49826

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Disable tracer warning for slicing indices."


Summary: If the index that is supplied from python is an integral type, it converts everything to int64_t which is traced correctly.

Test Plan: new test case

Reviewers:

Subscribers:

Tasks: https://github.com/pytorch/pytorch/issues/49826

Tags:

Differential Revision: [D25930773](https://our.internmc.facebook.com/intern/diff/D25930773)

[ghstack-poisoned]
